--- a/drivers/usb/serial/usb_wwan.c
+++ b/drivers/usb/serial/usb_wwan.c
@@ -470,7 +470,14 @@ static struct urb *usb_wwan_setup_urb(st
 	usb_fill_bulk_urb(urb, serial->dev,
 			  usb_sndbulkpipe(serial->dev, endpoint) | dir,
 			  buf, len, callback, ctx);
-
+	 #if 1
+	if(dir == USB_DIR_OUT) {
+		struct usb_device_descriptor *desc = &serial->dev->descriptor;
+		if(desc->idVendor == cpu_to_le16(0x2c7c)) {
+				urb->transfer_flags |= URB_ZERO_PACKET;
+		}
+	}
+	#endif
 	return urb;
 }
 
